src/nvim/edit.c

```diff
@@ -3825,6 +3831,21 @@ static void ins_compl_insert(void)
     compl_used_match = FALSE;
   else
     compl_used_match = TRUE;
+
+  // Set completed item.
+  // { word, abbr, menu, kind, info }
+  dict_T *dict = dict_alloc();
+  dict_add_nr_str(dict, "word", 0L,
+                  SHOULD_BE_STR(compl_shown_match->cp_str));
+  dict_add_nr_str(dict, "abbr", 0L,
+                  SHOULD_BE_STR(compl_shown_match->cp_text[CPT_ABBR]));
+  dict_add_nr_str(dict, "menu", 0L,
+                  SHOULD_BE_STR(compl_shown_match->cp_text[CPT_MENU]));
+  dict_add_nr_str(dict, "kind", 0L,
+                  SHOULD_BE_STR(compl_shown_match->cp_text[CPT_KIND]));
+  dict_add_nr_str(dict, "info", 0L,
+                  SHOULD_BE_STR(compl_shown_match->cp_text[CPT_INFO]));
```

Comment:
Unfortunatelly, I cannot fix it.
`v:completed_item` is readonly.
But each items are not.

di_flags does not work.